velero-2-mixin/targets.libsonnet

```diff
@@ -100,5 +100,115 @@ local utils = commonlib.utils {
+
+    backupSize:
+      prometheusQuery.new(
+        '${' + vars.datasources.prometheus.name + '}',
+        'velero_backup_tarball_size_bytes{%(queriesSelector)s}' % vars
+      )
+      + prometheusQuery.withLegendFormat('%s' % utils.labelsToPanelLegend(this.config.legendLabels)),
+
+    backupTime:
+      prometheusQuery.new(
+        '${' + vars.datasources.prometheus.name + '}',
+        'sum(increase(velero_backup_duration_seconds_bucket{le!="+Inf"}[$__interval:])) by (le)' % vars
+      ),
+
+
+    restoreSuccess:
+      prometheusQuery.new(
+        '${' + vars.datasources.prometheus.name + '}',
+        'increase(velero_restore_success_total{%(queriesSelector)s}[$__interval:])' % vars
+      )
+      + prometheusQuery.withLegendFormat('success'),
+
+    restoreFailure:
+      prometheusQuery.new(
+        '${' + vars.datasources.prometheus.name + '}',
+        'increase(velero_restore_failed_total{%(queriesSelector)s}[$__interval:])' % vars
+      )
+      + prometheusQuery.withLegendFormat('failure'),
+
+    restoreSuccessRate:
+      prometheusQuery.new(
+        '${' + vars.datasources.prometheus.name + '}',
+        'rate(velero_restore_success_total{%(queriesSelector)s}[$__rate_interval]) / clamp_min(rate(velero_restore_attempt_total{%(queriesSelector)s}[$__rate_interval]),0.001)' % vars
+      )
+      + prometheusQuery.withLegendFormat('%s' % utils.labelsToPanelLegend(this.config.legendLabels)),
+
+    volumeSnapshotSuccess:
+      prometheusQuery.new(
+        '${' + vars.datasources.prometheus.name + '}',
+        'increase(velero_volume_snapshot_success_total{%(queriesSelector)s}[$__interval:])' % vars
+      )
+      + prometheusQuery.withLegendFormat('success'),
+
+    volumeSnapshotFailure:
+      prometheusQuery.new(
+        '${' + vars.datasources.prometheus.name + '}',
+        'increase(velero_volume_snapshot_failure_total{%(queriesSelector)s}[$__interval:])' % vars
+      )
+      + prometheusQuery.withLegendFormat('failure'),
+
+    volumeSnapshotSuccessRate:
+      prometheusQuery.new(
+        '${' + vars.datasources.prometheus.name + '}',
+        'rate(velero_volume_snapshot_success_total{%(queriesSelector)s}[$__rate_interval]) / clamp_min(rate(velero_volume_snapshot_attempt_total{%(queriesSelector)s}[$__rate_interval]),0.001)' % vars
+      )
+      + prometheusQuery.withLegendFormat('%s' % utils.labelsToPanelLegend(this.config.legendLabels)),
+
+    csiSnapshotSuccess:
+      prometheusQuery.new(
+        '${' + vars.datasources.prometheus.name + '}',
+        'increase(velero_csi_snapshot_success_total{%(queriesSelector)s}[$__interval:])' % vars
+      )
+      + prometheusQuery.withLegendFormat('success'),
+
+    csiSnapshotFailure:
+      prometheusQuery.new(
+        '${' + vars.datasources.prometheus.name + '}',
+        'increase(velero_csi_snapshot_success_total{%(queriesSelector)s}[$__interval:])' % vars
+      )
+      + prometheusQuery.withLegendFormat('failure'),
```

Comment:
The time series graphs look very choppy for most of the left side panels that do not have a count. Is there some interpolation that should be done to smooth it out and make it look more like a timeseries?

Reply:
I was using the bar style of the time series which is why it looked like that. Looking back it's probably best to stick with the timeseries that we are used to.